agent/config/runtime.go

```diff
@@ -1411,16 +1404,18 @@ type RuntimeConfig struct {
 	// hcl: limits { txn_max_req_len = uint64 }
 	TxnMaxReqLen uint64
 
-	// UIDir is the directory containing the Web UI resources.
-	// If provided, the UI endpoints will be enabled.
+	// UIConfig holds various runtime options that control both the agent's
+	// behavior while serving the UI (e.g. whether it's enabled, what path it's
+	// mounted on) as well as options that enable or disable features within the
+	// UI.
 	//
-	// hcl: ui_dir = string
-	// flag: -ui-dir string
-	UIDir string
-
-	//UIContentPath is a string that sets the external
-	// path to a string. Default: /ui/
-	UIContentPath string
+	// NOTE: Never read from this field directly once the agent has started up
+	// since the UI config is reloadable. The on in the agent's config field may
+	// be out of date. Use the agent.getUIConfig() method to get the latest config
+	// in a thread-safe way.
+	//
+	// hcl: ui_config { ... }
+	UIConfig UIConfig
```

Comment:
The runtime reference on the agent is updated on a reload, so wouldn't it be safe to read it from this field as well?

Reply:
Ok, I am wrong about the main config reference. The current `ReloadConfig` is a bit confusing because it generates a new `consul.Config` based on `a.config`, which is not updated... It's not clear to me if `delegate.ReloadConfig` actually does anything. It seems like all it does is pass the same original config to the Client and Server. Maybe I am still missing something.

Reply:
I cleaned this up a bit in #8696